csrbuilder/__init__.py

```diff
@@ -162,19 +162,19 @@ def subject_public_key(self, value):
         object of the subject's public key.
         """
 
-        is_oscrypto = isinstance(value, asymmetric.PublicKey)
-        if not isinstance(value, keys.PublicKeyInfo) and not is_oscrypto:
-            raise TypeError(_pretty_message(
-                '''
-                subject_public_key must be an instance of
-                asn1crypto.keys.PublicKeyInfo or oscrypto.asymmetric.PublicKey,
-                not %s
-                ''',
-                _type_name(value)
-            ))
-
-        if is_oscrypto:
-            value = value.asn1
+        if not isinstance(value, keys.PublicKeyInfo):
+            from oscrypto import asymmetric
+            if isinstance(value, asymmetric.PublicKey):
+                value = value.asn1
+            else:
+                raise TypeError(_pretty_message(
+                    '''
+                    subject_public_key must be an instance of
+                    asn1crypto.keys.PublicKeyInfo or oscrypto.asymmetric.PublicKey,
+                    not %s
+                    ''',
+                    _type_name(value)
+                ))
```

Comment:
This change is simply inverting the logic, to first look for an asn1crypto.PublicKeyInfo, only if that fails should we import oscrypto.asymmetric.